.scripts/lib.sh

```diff
@@ -0,0 +1,92 @@
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+#
+
+
+# -----------------------------------------------------------------------------
+#  FUNCTIONS & PROCEDURES
+# -----------------------------------------------------------------------------
+
+function die() {
+    echo "$@" 1>&2 ; exit 1;
+}
+
+# Given a tag and a module name,
+#   if the tag is a release tag for module, echoes a tag
+#   else returns 1
+# Uses shopt and sets locale, so runs in subshell
+function is_release_tag() (
+    [ "$#" -eq 2 ] || die "2 arguments required, $# provided to 'func:is_release_tag'"
+    local __tag="$1"
+    local __module="$2"
+
+    # For extended glob patterns and ensure [0-9] range
+    shopt -s extglob
+    LC_ALL=C
+
+    case "${__module}" in
+        testing/micro)
+            tag_pattern='micro-+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag#micro-}" || return 1
+            ;;
+        testing/mini)
+            tag_pattern='+([0-9]).+([0-9]).+([0-9])/@(aws|gcp)/@(large|xlarge|xxlarge)'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag%%/*}" || return 1
+            ;;
+        1-trackers/golang-tracker)
+            tag_pattern='v+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag#v}" || return 1
+            ;;
+        4-storage/snowplow-rdb-loader)
+            tag_pattern='r+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag}" || return 1
+            ;;
+        5-data-modeling/data-models)
+            tag_pattern='**/+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag}" || return 1
+            ;;
+        *)
+            tag_pattern='+([0-9]).+([0-9]).+([0-9])'
+            [[ "${__tag}" = $tag_pattern ]] && echo "${__tag}" || return 1
+            ;;
+    esac
+)
+
+# Propagates a bump merge to the superproject
+function merge_bump() {
+    [ "$#" -eq 5 ] || die "5 arguments required, $# provided to 'func:merge_n_bump'"
+    local __top_level="$1"
+    local __mod_name="$2"
+    local __rel_path="$3"
+    local __bump_commit="$4"
+    local __bump_tag="$5"
+
+    # 1. merge in submodule
+    git merge --ff "${__bump_commit}"
+
+    # 2. go toplevel
+    cd "${__top_level}"
+
+    # 3. get repo name for commit message
+    repo_url=`git submodule--helper config submodule."${__mod_name}".url`
+    [ -z "${repo_url}" ] && die "Submodule url not found for ${__mod_name}"
+    repo_name=`basename "${repo_url}" .git`
+    commit_msg="${repo_name}: Bump to ${__bump_tag}"
```

Comment:
I would replace `Bump to` with `Release`

Reply:
Thank you for reviewing @benjben !
The reason i did it this way was mainly to stay consinstent with the commits' messages of the manual updates. I can certainly change it to `Release`. Do you think it would be preferred from the point of view of the `snowplow/snowplow` repository?

Reply:
I think that `Release` sends a stronger message than `Bump`, but that might be just a matter of taste.